table/table.go

```diff
@@ -741,7 +740,61 @@ func (table *Table) InitRoutes(config cfg.Config, meta toml.MetaData) error {
 
 			route, err := route.NewKafkaRoute(routeConfig.Key, routeConfig.Prefix, routeConfig.Substr, routeConfig.Regex, writerConfig, routingMutator)
 			table.AddRoute(route)
+		case "bg_metadata":
+			bgMetadataCfg := routeConfig.BgMetadata
+			if bgMetadataCfg == nil {
+				return fmt.Errorf("error adding route '%s': bg_metadata config is not specified", routeConfig.Key)
+			}
+			if bgMetadataCfg.ShardingFactor == 0 {
+				return fmt.Errorf("error adding route '%s': sharding factor must be specified", routeConfig.Key)
+			}
+			if bgMetadataCfg.FilterSize == 0 {
+				return fmt.Errorf("error adding route '%s': filter size must be specified", routeConfig.Key)
+			}
+			if bgMetadataCfg.FaultTolerance == 0 {
+				return fmt.Errorf("error adding route '%s': fault tolerance percentage must be specified", routeConfig.Key)
+			}
+			if bgMetadataCfg.FaultTolerance < 0 || bgMetadataCfg.FaultTolerance > 1 {
+				return fmt.Errorf("error adding route '%s': fault tolerance value must be between 0 and 1", routeConfig.Key)
+			}
+
+			if bgMetadataCfg.ClearInterval == "" {
+				return fmt.Errorf("error adding route '%s': clear interval value must be specified", routeConfig.Key)
+			}
+
+			clearInterval, err := time.ParseDuration(bgMetadataCfg.ClearInterval)
+			if err != nil {
+				return fmt.Errorf("error adding route '%s': could not parse clear_interval", routeConfig.Key)
+			}
+
+			clearWait, err := time.ParseDuration(bgMetadataCfg.ClearWait)
+			if err != nil && bgMetadataCfg.ClearWait != "" {
```

Comment:
|| ?

Reply:
It's intentional since the value can be empty

Reply:
Yes, but if err is nil, it doesn't matter if its empty, it's still an error

Reply:
It's still relevant to check for error then to check for invalid value

err != nil should be enough to trigger error behavior

Reply:
It does trigger an error, yes, but for a case that I don't want. To be parsable (not get an error):
1) there has to at least be some value **AND**
2) the value has to be a time interval
If there is no value, parsing will return a non nil error value which will not allow the interval to be unspecified.